src/Pulsar.Client/Auth/Oauth2/AuthenticationOauth2.fs

```diff
@@ -81,6 +80,32 @@ type internal AuthenticationOauth2(issuerUrl: Uri, audience: string, privateKey:
             return temp
         }
 
+    let base64DeserializeCreds data =
+        let bytes = Convert.FromBase64String(data)
+        let decodedData = System.Text.Encoding.UTF8.GetString(bytes)
+        JsonSerializer.Deserialize<Credentials>(decodedData)
+
+    let decodeAndDeserializeCreds (privateKey: Uri) =
+        let split = privateKey.LocalPath.Split(',', 2)
```

Comment:
I would write it in this way:
 ```fsharp
 match privateKey.LocalPath.Split(',', 2) with
 | [| contentType, data |] when contentType = "application/json;base64" ->
     base64DeserializeCreds data
 | _ ->
     raise <| NotSupportedException $"Content type '{contentType}' is not supported."
 ```

Reply:
Done, I had to add an extra case for when there are a different number of elements in the split